text/0814-stylelint.md

```diff
@@ -0,0 +1,138 @@
+* [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node) - Node Best Practices
+* [eslint-plugin-qunit](https://github.com/platinumazure/eslint-plugin-qunit) - QUnit Best Practices [RFC](https://github.com/emberjs/rfcs/blob/master/text/0702-eslint-plugin-qunit.md)
+* [prettier](https://prettier.io/) - Automated Code Styling [RFC](https://github.com/emberjs/rfcs/blob/master/text/0628-prettier.md)
+
+With all these amazing tools in place, however, we still have a major aspect of our Ember codebases 
+that are not benefiting from mature tooling to enforce best practices and automate code styling: CSS.
+
+[Stylelint](https://github.com/stylelint/stylelint) is a modern linter for stylesheets that ships with 
+over 170 built-in rules, sane default configs, and robust plugin support.
+
+## Detailed design
+
+The general idea is that we will update the `app` and `addon` [blueprints](https://github.com/ember-cli/ember-cli/tree/master/blueprints) 
+to add a few Stylelint-related packages to the `package.json`, add config files to enable this tooling 
+by default, and add commands to the `scripts` section of `package.json` to expose easy access to this 
+tool consistent with our existing linters.
+
+### Packages
+
+The following packages will be added to the `devDependencies` section of both blueprints:
+
+* [stylelint](https://github.com/stylelint/stylelint) - The base linter
+* [stylelint-config-standard](https://github.com/stylelint/stylelint-config-standard) - The standard rule config supported by the stylelint community
+* [stylelint-prettier](https://github.com/prettier/stylelint-prettier) - Plugin to support prettier auto-formatting as stylelint rules
+* [stylelint-config-prettier](https://github.com/prettier/stylelint-config-prettier) - Plugin to disable stylelint rules that conflict with prettier's automatic behavior
+
+Prettier itself is already included in the Ember blueprints by default.
+
+### Configuration Changes
+
+A `.stylelintrc.js` file will be added with the following contents:
+
+```js title=".stylelintrc.js"
+'use strict';
+
+module.exports = {
+  extends: ['stylelint-config-standard', 'stylelint-prettier/recommended'],
+};
+```
+
+There are many rules, configs, and plugins available for stylelint and prettier, but we are 
+intentionally avoiding being prescriptive on anything that isn't nearly universally agreed upon 
+by the stylelint, prettier, and Ember communities.
+
+### Blueprint Changes
+
+In general, it is recommended that all blueprints provided by addons should satisfy the default 
+linting configuration of a new Ember application. As such the blueprints provided by ember-cli, 
+ember-source, and ember-data will be updated as needed to ensure that they satisfy these new 
+linting rules requirements.
+
+#### `package.json` scripts
+
+The `app` and `addon` blueprints will be updated to add style-related linting scripts:
+
+```json title="package.json"
+{
+  "scripts": {
+    "lint:style": "stylelint .",
```

Comment:
Good callout. I think this is because Stylelint supports CSS-in-JS, which we probably don't need to support in our default configuration. An ignore file would likely be difficult to maintain, so we can scope this call to the most common filetypes: CSS, SCSS, SASS, LESS

Reply:
Or, should we explicitly only support CSS with the default config since that's what Ember apps are configured for, even though stylelint can support other file types without additional plugins/configuration?

Reply:
The stylelint config suggested in this RFC only works for plain `.css` files afaict, so including `scss`, `sass` and `less` seems unnecessary to me. If you do try to lint e.g. a `scss` file with the suggested config, you'd get:

```
When linting something other than CSS, you should install an appropriate syntax, e.g. "postcss-scss", and use the "customSyntax" option
```

Reply:
I still think we either need to make the path even more specific or introduce a `.stylelintignore` file (similar to `.eslintignore` and `.prettierignore`). At the moment, CSS files in the following folders would be linted as well:
- `coverage`
- `vendor`
- `dist`
- `ember-try` folders
- ...

Reply:
I agree that we should probably do both - we'll need the stylelintignore but also should scope it to css files

Reply:
I've restricted the scope to css file types and added a lint ignore section based on the existing ignore files.

Reply:
It's possible the suggested ignore file is slightly broader than it needs to be, but I'd favor consistency with other ignore files over removing something just because we don't *expect* it to ever contain CSS files.